src/api/routes/user-routes.ts

```diff
@@ -1,16 +1,20 @@
 import express from 'express';
 import { userController } from '../controllers';
+import { validate } from '../middlewares/validateResource';
+import * as userSchema from '../schemas/user-schema';
 
 const router = express.Router();
 
-router.get('/user/get', userController.getAllUsers);
+//  Throw proper error when req.body has syntax errors.
 
-router.get('/user/getUserByEmail', userController.getUserByEmail);
+router.get('/user/getAllUsers', userController.getAllUsers);
+
+router.get('/user/getUserByEmail', validate(userSchema.getUserByEmailSchema), userController.getUserByEmail);
 
 router.post('/user/createSampleUser', userController.createSampleUser);
 
-router.post('/user/createSimpleUser', userController.createSimpleUser);
+router.post('/user/createSimpleUser', validate(userSchema.createUserSchema), userController.createSimpleUser);
 
-router.post('/user/updateUserProfile', userController.updateUserProfile);
+router.post('/user/updateUserProfile', validate(userSchema.updateUserSchema), userController.updateUserProfile);
```

Comment:
The update user profile route should probably be a PUT request rather than POST

Reply:
Great remark. I will update that. PUT is indead more appropriate since they are no side effects to calling the same request multiple times.